### PR TITLE
ui: Make peered intentions read-only

### PR DESCRIPTION
**Pull request by johncowen**
### Description
Peered intentions are currently readonly. Clicking the rows will now take you to a readonly detail page based on the fact that intentions with SourcePeer set are not writable.

### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

/cc @LevelbossMIke

### CODE REVIEW

File: ui/packages/consul-ui/app/abilities/intention.js [hunk 1]
```diff
@@ -4,7 +4,12 @@ export default class IntentionAbility extends BaseAbility {
   resource = 'intention';
 
   get canWrite() {
-    return super.canWrite && (typeof this.item === 'undefined' || !this.canViewCRD);
+    // Peered intentions aren't writable
+    if(typeof this.item !== 'undefined' && typeof this.item.SourcePeer !== 'undefined') {
```
Comment (evrowe) [inline ui/packages/consul-ui/app/abilities/intention.js:8]: The changes to the the intentions model below renames `SourcePeer` to `PeerName`; is this logic still accurate? Or does the reference from `this.item.SourcePeer` need to change to `this.item.PeerName`?

If `SourcePeer` is still correct, where does that come from?

File: ui/packages/consul-ui/tests/acceptance/dc/intentions/delete.feature [hunk 2]
```diff
@@ -4,6 +4,7 @@ Feature: dc / intentions / deleting: Deleting items with confirmations, success
     Given 1 datacenter model with the value "datacenter"
     And 1 intention model from yaml
     ---
+    SourcePeer: ""
```
Comment (evrowe) [inline ui/packages/consul-ui/tests/acceptance/dc/intentions/delete.feature:7]: Same question as above, is this still accurate?

**Review by evrowe — COMMENTED** [review]
couple of quick clarifying questions

**Comment by johncowen** [comment]
Ooohhhh copypasta fail 😆 , fixed up here https://github.com/hashicorp/consul/pull/13814/commits/048e2bba7f7af0fc11b76ac08336951abd51b8da thanks for the spot! That should answer both your queries here.

**Review by evrowe — APPROVED** [review]
Nice, that last change addresses my questions

**Pull request merged**